lunrjs Search

ashfurrow opened the pull request:
Fixes https://github.com/artsy/artsy.github.io/issues/329

So we're using [a fork](https://github.com/ashfurrow/jekyll-lunr-js-search/tree/v2) of [this library](https://github.com/slashdotdash/jekyll-lunr-js-search), because the library requires Jekyll 3 (as far as I can tell) and there are no semantic tags available to reference in our Gemfile. I [added Jekyll 2-style posts](https://github.com/ashfurrow/jekyll-lunr-js-search/commit/367d165be6db8fa1440a9d95dda18527c98063d1#diff-7afa659b015f4715a4978fc3949eba71R135), and there appeared to be some weirdness with [emoji in Jekyll](https://github.com/olivernn/lunr.js/issues/243) being index in lunr.js so I filtered them out of the index [in the first way that Google suggested](https://github.com/ashfurrow/jekyll-lunr-js-search/commit/f39837bf2fc7a1b2ac408b2d1386acefbb65c219) which looks terrible but works.

The design needs a little work for sure. Here's a before/after shot:

![screen shot 2017-03-25 at 7 29 31 pm](https://cloud.githubusercontent.com/assets/498212/24327004/ca272e9e-1192-11e7-9d36-ef2550707add.png)
![screen shot 2017-03-25 at 7 28 40 pm](https://cloud.githubusercontent.com/assets/498212/24327005/ca27782c-1192-11e7-885c-0eab69beaed0.png)

The downside compared to Google Site search is that we don't get a preview of the post, due to limitations in lunr.js. I think that's okay, but wanted to surface it for discussion.

orta commented:
Looks perfect - thanks! Only a design request change, copy the design of the archives - http://artsy.github.io/blog/archives/

and instead of the year have something like "search results for Eigen"

Then this is a 👍

ashfurrow commented:
Cool, updated this. Had to update the fork, added some documentation there about the modifications I've made. Here's a preview:

![screen shot 2017-03-26 at 11 39 03 am](https://cloud.githubusercontent.com/assets/498212/24332678/fb17cb40-1218-11e7-9e27-3278811f0fa7.png)

There are no results displayed while the search index is being downloaded (~3MB) so we should verify the UX is okay after deploying. Oh, and the search box _on the search page_ is search-as-you-type, which is pretty cool.

orta commented:
Alright, looks good - nice work on the README for your fork 👍
